### PR TITLE
Support test project building and running against Windows Desktop fra…

### PR DESCRIPTION
**Pull request by tarekgh**
The changes here is to add the build configuration to the globalization test project to allow it to compile and run against the desktop framework. 
The changes also include changing the test project.json to add the desktop runtime support also we have upgraded to use the latest version of the buildtools

### CODE REVIEW

**Comment by tarekgh** [comment]
@JeremyKuhne @weshaggard 

this is the corefx to consume the changes done in buildtools

File: src/Common/test-runtime/project.json [hunk 1]
```diff
@@ -1,30 +1,59 @@
...
+        "win7-x64": {},
+        "win7-x86": {},
+        "ubuntu.14.04-x64": {},
+        "osx.10.10-x64": {},
+        "centos.7-x64": {},
+        "rhel.7-x64": {},
+        "debian.8.2-x64": {}
+      }
+    }, 
+    "net46": {
```
Comment (weshaggard) [inline src/Common/test-runtime/project.json]: Can we get away with just the lowest version? net45?

File: src/System.Globalization/tests/System.Globalization.Tests.csproj [hunk 2]
```diff
@@ -11,10 +11,23 @@
...
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' " />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net462_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net462_Release|AnyCPU'" />
+
+  <PropertyGroup>
+    <IsWindowsDesktop Condition="'$(IsWindowsDesktop)' =='' And '$(TargetGroup)' == 'net45'">true</IsWindowsDesktop>
```
Comment (weshaggard) [inline src/System.Globalization/tests/System.Globalization.Tests.csproj]: We should use the TargetGroup name based on the convention. So the define should be `net45`.

File: src/System.Globalization/tests/RegionInfo/RegionInfoTests.Properties.cs [hunk 3]
```diff
@@ -12,7 +12,12 @@ public class RegionInfoPropertyTests
         [Fact]
         public void CurrentRegion()
         {
+#if WindowsDesktop
```
Comment (weshaggard) [inline src/System.Globalization/tests/RegionInfo/RegionInfoTests.Properties.cs]: All these #ifdefs tell me that we actually have a lot of inconsistencies between the target frameworks we should actually take a close look at these to be sure we are OK with these differences.

**Comment by tarekgh** [comment]
@weshaggard I have addressed all your comments. please have a quick look and let me know if your have any more comments

File: src/System.Globalization/tests/System.Globalization.Tests.csproj [hunk 4]
```diff
@@ -11,10 +11,11 @@
     <IncludePerformanceTests>true</IncludePerformanceTests>
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
```
Comment (weshaggard) [inline src/System.Globalization/tests/System.Globalization.Tests.csproj]: nit: can you please delete the space at the beginning and end of these conditions.

File: src/System.Globalization/tests/System.Globalization.Tests.csproj [hunk 5]
```diff
@@ -140,8 +141,19 @@
       <Link>Common\System\RandomDataGenerator.cs</Link>
     </Compile>
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)' == 'net46'">
```
Comment (weshaggard) [inline src/System.Globalization/tests/System.Globalization.Tests.csproj]: Don't you still need the #ifdef's?

File: src/System.Globalization/tests/System.Globalization.Tests.csproj [hunk 6]
```diff
@@ -140,8 +141,19 @@
...
+    <TargetingPackReference Include="System.Threading.Tasks" /> 
+    <TargetingPackReference Include="System.Core" /> 
+    <TargetingPackReference Include="System.Runtime" />
+  </ItemGroup>  
   <ItemGroup>
     <None Include="project.json" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+
+  <PropertyGroup>
```
Comment (weshaggard) [inline src/System.Globalization/tests/System.Globalization.Tests.csproj]: Can you please file an issue in BuildTools on this so we get it fixed and not propagate this in all the project configurations.

File: src/System.Globalization/tests/System.Globalization.Tests.csproj (hunk 5 above)
Comment (tarekgh) [inline src/System.Globalization/tests/System.Globalization.Tests.csproj]: No. I handled the case in the test code itself.

File: src/System.Globalization/tests/System.Globalization.Tests.csproj (hunk 6 above)
Comment (tarekgh) [inline src/System.Globalization/tests/System.Globalization.Tests.csproj]: done! 
https://github.com/dotnet/buildtools/issues/596

**Comment by tarekgh** [comment]
https://github.com/dotnet/buildtools/issues/596
@weshaggard all done. thanks for your review.

**Pull request merged**